### PR TITLE
Add a Knative demo application

### PR DESCRIPTION
**Pull request by zzxwill**
This app demonstrates how OAM helps Knative Configuration/Route
to  interact with cloud resources provisioned by Crossplane.

### CODE REVIEW

File: 6.Knative_with_Cloud_Resource_App/README.md [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+              traffic:
+                - revisionName: webapp-8t2h5
+                  percent: 100
+
+    - componentName: db
+```
+
+Check the status of Knative Route.
+```shell
+✗ k get route
```
Comment (wonderflow) [inline 6.Knative_with_Cloud_Resource_App/README.md:221]:
```suggestion
$ kubectl get route
```

File: 6.Knative_with_Cloud_Resource_App/README.md [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+NAME             READY   SYNCED   STATE      ENGINE       VERSION   AGE
+rdspostgresql3   True    True     Running    PostgreSQL   9.4       11m
+```
+At this time, a Postgresql instance is provisioned, and we can see it from Alibaba Clod console.
+![image.png](./postgresql_instance.png)<br />
+
+The application `webapp` also generates a Knative Configuration.
+
+```shell
+✗ k get configuration
```
Comment (wonderflow) [inline 6.Knative_with_Cloud_Resource_App/README.md:189]:
```suggestion
$ kubectl get configuration
```

File: 6.Knative_with_Cloud_Resource_App/README.md [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+By checking the list of Component, two components `db` and `webapp` are provisioned. After several minutes,
+the statue of RDSInstance will become to `Running`.
+```shell
+✗ k get component
+NAME        WORKLOAD-KIND
+db          RDSInstance
+webapp      Configuration
+
+✗ k get RDSInstance
```
Comment (wonderflow) [inline 6.Knative_with_Cloud_Resource_App/README.md:179]:
```suggestion
$ kubectl get RDSInstance
```

File: 6.Knative_with_Cloud_Resource_App/README.md [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+component.core.oam.dev/webapp created
+
+✗ kubectl apply -f appconfig-1.yaml
+applicationconfiguration.core.oam.dev/knative-postgresql-appconfig configured
+```
+
+By checking the list of Component, two components `db` and `webapp` are provisioned. After several minutes,
+the statue of RDSInstance will become to `Running`.
+```shell
+✗ k get component
```
Comment (wonderflow) [inline 6.Knative_with_Cloud_Resource_App/README.md:174]:
```suggestion
$ kubectl  get component
```

File: 6.Knative_with_Cloud_Resource_App/README.md [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+
+The application `webapp` also generates a Knative Configuration.
+
+```shell
+✗ k get configuration
+NAME     LATESTCREATED   LATESTREADY    READY   REASON
+webapp   webapp-c8s4n    webapp-c8s4n   True
+```
+
+Edit `appconfig-1.yaml`, add Knative Route with the latest revision of Configuration as shown in `appconfig-2.yaml` and submit it.
```
Comment (wonderflow) [inline 6.Knative_with_Cloud_Resource_App/README.md:194]: This step makes the demo break. We need dependency feature \cc @hongchaodeng

File: 6.Knative_with_Cloud_Resource_App/Components/component-knative-configuration-flask-app.yaml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    apiVersion: serving.knative.dev/v1
+    kind: Configuration
+    metadata:
+      name: webapp
+      namespace: default
+    spec:
+      template:
+        spec:
+          containers:
+            - image: zzxwill/postgresql-flask-web-application:v0.1
```
Comment (ryanzhang-oss) [inline 6.Knative_with_Cloud_Resource_App/Components/component-knative-configuration-flask-app.yaml:16]: better to use oamdev account

Comment (zzxwill) [reply 6.Knative_with_Cloud_Resource_App/Components/component-knative-configuration-flask-app.yaml:16]: Sure, would you please help grant me with write authority?

File: 6.Knative_with_Cloud_Resource_App/README.md (hunk 5 above)
Comment (zzxwill) [reply 6.Knative_with_Cloud_Resource_App/README.md:194]: Will use datainput/dataoutput mechanism after the [PR](https://github.com/crossplane/oam-kubernetes-runtime/pull/48) is merged.

Comment (zzxwill) [reply 6.Knative_with_Cloud_Resource_App/README.md:194]: Please hold on the code merging, I hit an issue for verifying this App with the pending PR https://github.com/crossplane/oam-kubernetes-runtime/pull/48#issuecomment-648655426.

**Comment by wonderflow** [comment]
I think we can merge it, and improve this demo after dependency is ready.

**Comment by wonderflow** [comment]
Can anyone with write access help merge it? @suhuruli @vturecek @technosophos

**Comment by hongchaodeng** [comment]
lgtm

**Pull request merged**